### PR TITLE
fix(tools/scripts): convert more tests to TypeScript

### PR DESCRIPTION
**Pull request by sam-rice**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Contributing to [#48753](https://github.com/freeCodeCamp/freeCodeCamp/issues/48753)

Hello! This is my first time contributing, and I wanted to get a smaller PR here to confirm that I'm approaching these conversions correctly. I'm happy to continue converting a larger batch of specs to be included in this PR, if there isn't enough here to justify a merge.

~~**All changed tests were verified as passing when ran locally.**~~ see comment below

### Files converted:

- challenge-hot-keys.ts
  - selectors variable name changed due to scoping issue
- output.ts
  - selectors and locations variable names changed due to scoping issues
- searchBar.ts
  - assertion for URL changed to test for [expected behavior described here](https://github.com/freeCodeCamp/freeCodeCamp/pull/48776#issuecomment-1357277784). I wasn't able to get the exact code that @scissorsneedfoodtoo recommended to pass.

### CODE REVIEW

**Comment by sam-rice** [comment]
Hi, I'm having an issue getting the search-bar.ts test to pass on all three browsers. Apparently I was only testing for Electron when I originally ran each test individually from the command line. I'm not sure if I need to write a better assertion, or if I'm misunderstanding the way the tests work. Any guidance would be appreciated. See screenshots:

### On my original approach, FIrefox/Chrome are failing here, but Electron is passing:
Screen Shot 2023-01-24 at 9 37 39 AM

### Making the change for Firefox/Chrome causes the Electron test to fail:
Screen Shot 2023-01-24 at 9 30 19 AM

**Comment by Sboonny** [comment]
It was stuck on infinite loop, Naomi sorted it out

**Review by naomi-lgbt — APPROVED** [review]
banger pr, thanks for contributing!

File: cypress/e2e/default/learn/common-components/search-bar.ts [hunk 1]
```diff
@@ -106,6 +106,6 @@ describe('Search bar', () => {
 
     cy.get('.ais-SearchBox-form').submit();
 
-    cy.url('/');
+    cy.url().should('eq', 'http://localhost:8000/learn/');
```
Comment (ShaunSHamilton) [inline cypress/e2e/default/learn/common-components/search-bar.ts:109]:
```suggestion
    cy.location().should(loc => {
      expect(loc.pathname).to.eq('/learn');
    });
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thank you for taking the time to contribute.

We (mostly me) often run these tests in non-local environments (Gitpod). So, tests that assert locations like `localhost` cause that to fail.

This should be an acceptable alternative - I stole it from another test 🤷‍♂️

File: cypress/e2e/default/learn/common-components/search-bar.ts (hunk 1 above)
Comment (sam-rice) [reply cypress/e2e/default/learn/common-components/search-bar.ts]: Cool, that worked when testing on my machine 👍

Side question: As a general approach, is it sufficient to run tests like `npm run cypress -- run --spec=cypress/e2e/default/learn/common-components/search-bar.ts`, or should the entire test suite be ran before submitting a PR? (`npm run cypress:dev:run`)

Comment (ShaunSHamilton) [reply cypress/e2e/default/learn/common-components/search-bar.ts]: If you are changing test files, it is sufficient to run just the specs you are changing.

If you are changing client/server code, it is best to run the whole suite (at some point).

File: cypress/e2e/default/learn/common-components/search-bar.ts [hunk 2]
```diff
@@ -106,6 +106,8 @@ describe('Search bar', () => {
 
     cy.get('.ais-SearchBox-form').submit();
 
-    cy.url('/');
+    cy.location().should(loc => {
+      expect(loc.pathname).to.eq('/learn');
```
Comment (ShaunSHamilton) [inline cypress/e2e/default/learn/common-components/search-bar.ts:110]: Sorry about that. This is finicky.

```suggestion
      expect(loc.pathname).to.eq('/learn/');
```

**Review by ShaunSHamilton — APPROVED** [review]
Thank you for making those changes. This LGTM 👍

**Comment by ShaunSHamilton** [comment]
Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**